pandas/core/nanops.py

```diff
@@ -1007,17 +1098,21 @@ def nanprod(values, axis=None, skipna=True, min_count=0, mask=None):
     --------
     The product of all elements on a given axis. ( NaNs are treated as 1)
     """
-    if mask is None:
-        mask = isna(values)
-    if skipna and not is_any_int_dtype(values):
+    mask = _maybe_get_mask(values, skipna, mask)
+
+    if skipna and not is_any_int_dtype(values) and mask is not None:
```

Comment:
do you need the int check here? (as the mask will by definition be None if we have ints)

Reply:
No. The odd case of `Series(dtype='bool').prod()` will be slightly faster as a result of it being removed.